rmw_zenoh_cpp/src/detail/zenoh_utils.hpp

```diff
@@ -92,6 +98,36 @@ class Payload
   // and `zenoh::Slice` plus a `zenoh::Bytes` otherwise.
   std::variant<NonContiguous, Contiguous, Empty> bytes_;
 };
+
+///=============================================================================
+class BufferPool
+{
+public:
+  struct Buffer
+  {
+    uint8_t * data;
+    size_t size;
+  };
+
+  BufferPool();
+
+  ~BufferPool();
+
+  Buffer allocate(size_t size);
```

Comment:
To make it clear that allocation failed as opposed to relying on checking `serialization_buffer.data == nullptr` in `rmw_publisher_data.cpp`.
```suggestion
  std::optional<Buffer> allocate(size_t size);
```

Reply:
Addressed in cefeedeacd53cc3d310bcd08605cd9fd4e3ca797